pkg/apis/batch/validation/validation.go

```diff
@@ -384,9 +386,40 @@ func validateTimeZone(timeZone *string, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(fldPath, timeZone, err.Error()))
 	}
 
+	if err := validateTimeZoneCase(*timeZone); err != nil {
```

Comment:
Is the reason we have the MacOS guard because we believe on other systems there may be a valid usecase for having case-insensitive timezone changes?

If all OS's can have case-insensitive filesystems, then this doesn't make sense.

Also, this is validation from the server process... that feels... very hacky, to put that check in here, since we don't typically have OS specific checks in server side code.

This also has the side effect of forcing the server process to only accept timezones in the GOROOT, not the ones in the system.  That feels wrong, and I am very uncomfortable having server code having different behavior.

What is the alternate fix for this in the test side?

Reply:
This is to fulfill the requirement we found after merging alpha, see https://github.com/kubernetes/enhancements/blob/master/keps/sig-apps/3140-TimeZone-support-in-CronJob/README.md#beta which points to both https://github.com/kubernetes/kubernetes/pull/109218 which added skip for a case-sensitive TZ on a mac, and https://github.com/golang/go/issues/21512 explaining why golang doesn't care about case in TZ database.

Reply:
I'm saying the requirement shouldn't be solved like this.

At a first read this code doesn't belong here (it's mac specific, not "insensitive filesystem specific", and it works around a problem by identifying the environment, rather than resolving the problem).  If you're on a Mac, and you provide "foo" to CronJob timezone, and the system database says "i've got foo", given how we've defined the field, we must accept that input.

The linked test is wrong - based on the defined behavior of the field, the database could have `AMERICA/new_york` in it, which means our validation automatically accepts whatever values are in that file.  In effect, we are not validating the locale name, we're consulting a random third party (the host fs) for info that usually, but not always, is consistent.  

I don't think this is something that should be fixed in validation.  This is instead something we try to verify via our tests - that a third party TZ data entry can be loaded based on the current environment of the process.  I.e. create a synthetic file on Mac, try to load it.  Create a synthetic timezone DB on Linux, set the process lookup so that `time.LoadLocation` checks that, verify we find the arbitrary thing we look for.